test/e2e/upgrade_test.go

```diff
@@ -35,6 +35,7 @@ func runSimpleUpgradeFlow(test *framework.ClusterE2ETest, updateVersion v1alpha1
 	test.ValidateCluster(updateVersion)
 	test.StopIfFailed()
 	test.DeleteCluster()
+	test.ValidateHardwareDecommissioned()
```

Comment:
Why do we need this here? This method wont be called for tinkerbell tests.

same for [line 49](https://github.com/aws/eks-anywhere/pull/3147/files#diff-fc8ea8c7c1a732fd9a425771cb3463d1cd143e5527acec751f32ba88f2df697aR49)

Reply:
test.ValidateHardwareDecommissioned() confirms DeleteCluster() worked as desired

Reply:
As @gwesterfieldjr mentioned, it makes sure that DeleteCluster decommissioned the hardware as expected.

Reply:
Right that's why we have it under `runSimpleUpgradeFlowForBareMetal` but these methods are used only for vsphere and other providers right? We dont need this check here.

Reply:
ahh good point!!

Reply:
oh facepalm, yes, thanks!